hbase-server/src/main/java/org/apache/hadoop/hbase/master/HMaster.java

```diff
@@ -707,6 +708,7 @@ protected MasterRpcServices createRpcServices() throws IOException {
   protected void configureInfoServer(InfoServer infoServer) {
     infoServer.addUnprivilegedServlet("master-status", "/master-status", MasterStatusServlet.class);
     infoServer.addUnprivilegedServlet("api_v1", "/api/v1/*", buildApiV1Servlet());
+    infoServer.addUnprivilegedServlet("hbck", "/hbck", MasterHbckServlet.class);
```

Comment:
Why add a new endpoint instead of having the existing `/hbck.jsp` honor an "application/json" header and render its response there? Probably a bunch of that jsp's POJOs can be reused...

Reply:
I did not and still do not know how to make a JSP page honor different content-encodings.